src/preprocessors/ember.ts

```diff
@@ -0,0 +1,63 @@
+import { PrettierOptions } from '../types';
+import { replaceAt } from '../utils/replace-at';
+import { preprocessor } from './preprocessor';
+
+const getParseImportsExports = () => {
+    try {
+        const parseImportsExports = require('parse-imports-exports');
+
+        return parseImportsExports;
+    } catch {
+        throw new Error(
+            'parse-imports-exports could not be loaded.  Is it installed?',
+        );
+    }
+};
+
+const sortImports = (code: string, options: PrettierOptions) => {
+    const { parseImportsExports } = getParseImportsExports();
+    const importsExports = parseImportsExports(code, {
+        ignoreDynamicImports: true,
+        ignoreRegexpLiterals: true,
+        ignoreRequires: true,
+        ignoreCommonJsExports: true,
+    });
+
+    let justImports = '';
+
+    function ingest(collection?: {
+        [key: string]: readonly {
+            readonly start: number;
+            readonly end: number;
+        }[];
+    }) {
+        if (!collection) return;
+
+        for (let [, info] of Object.entries({ ...collection })) {
+            for (let pos of info) {
```

Comment:
Nit: I prefer using `const` for variables that are not intended to be changed, rather than `let`.